solr/core/src/java/org/apache/solr/api/V2HttpCall.java

```diff
@@ -360,6 +366,80 @@ protected void execute(SolrQueryResponse rsp) {
     SolrCore.postDecorateResponse(handler, solrReq, rsp);
   }
 
+  @Override
+  protected void populateTracingSpan(Span span) {
+    // Set db.instance
+    String coreOrColName = this.origCorename;
+    if (coreOrColName == null) {
+      Map<String, String> pathTemplateValues = getUrlParts(); // == solrReq.getPathTemplateValues()
+      coreOrColName = pathTemplateValues.get("collection");
+      if (coreOrColName == null) {
+        coreOrColName = pathTemplateValues.get("core");
+      }
+    }
+    if (coreOrColName != null) {
+      span.setTag(Tags.DB_INSTANCE, coreOrColName);
+    }
+
+    // Get the templatize-ed path
+    String path;
+    if (api instanceof AnnotatedApi) {
+      // ideal scenario; eventually everything might be AnnotatedApi?
+      path = ((AnnotatedApi) api).getEndPoint().path()[0]; // consider first to be primary
+    } else {
+      path = computeEndpointPath();
+      // TODO consider getValidators, looking for command & path?
+    }
+
+    String verb = req.getMethod().toLowerCase(Locale.ROOT);
```

Comment:
why use to `toLowerCase(Locale.ROOT);`

it's expensive. it's always going to be in ALL CAPS

Reply:
It's common in tracing (from my limited experience) to use lower-case, so I'm trying to be consistent here.  Also I very much disagree on the relative cost of this compared to the limited cases where this is activated.

Reply:
is this used rarely or in every request?

Reply:
If someone opts-in to tracing, then every request is traced but not necessarily "reported" to a tracing server (maybe 1% depending on configuration).  Unfortunately even a non-reported span will be decorated with this metadata that is ultimately not used.  Ideally the tracing impl chosen can return a Span implementing NoopSpan which will avoid this logic here but neither Jaeger nor Brave OpenTracing adapters do this :-/  I don't know about others.  I reported this enhancement idea to Brave.

Reply:
BTW when I said 

> the relative cost of this compared to the limited cases where this is activated.

I not only meant opt-in to tracing but in the big picture of all that goes on to field a request into Solr to actually do something.  Lowercasing a string is inconsequential.